File: jvm/src/main/kotlin/org/faziodev/saywhatstarter/types/Response.kt
package org.faziodev.saywhatstarter.types

import org.faziodev.saywhatstarter.types.messages.Message

data class Response(
    val source: String,
    val fulfillmentText: String? = null,
    val fulfillmentMessages: List<Message> = listOf(),
    val payload: Map<String, String>? = mapOf(),
    val outputContexts: List<Context>? = listOf(),
    val followupEventInput: EventInput? = null
)

File: csharp/SayWhatStarterWebhook/Models/Response.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using SayWhatStarterWebhook.Converters;
using SayWhatStarterWebhook.Models.Messages;

namespace SayWhatStarterWebhook.Models
{
    public class Response
    {
        public string Source { get; set; }
        public string FulfillmentText { get; set; }
        [JsonProperty(ItemConverterType = typeof(MessageConverter))]
        public IList<IMessage> FulfillmentMessages { get; set; }
        public IDictionary<string, string> Payload { get; set; }
        public IList<Context> OutputContexts { get; set; }
        public EventInput FollowupEventInput { get; set; }
    }
}